op-challenger/game/fault/trace/outputs/split_adapter.go

```diff
@@ -16,42 +16,49 @@ type ProposalTraceProviderCreator func(ctx context.Context, localContext common.
 
 func OutputRootSplitAdapter(topProvider *OutputTraceProvider, creator ProposalTraceProviderCreator) split.ProviderCreator {
 	return func(ctx context.Context, depth types.Depth, pre types.Claim, post types.Claim) (types.TraceProvider, error) {
-		localContext := createLocalContext(pre, post)
-		usePrestateBlock := pre == (types.Claim{})
-		var agreed contracts.Proposal
-		if usePrestateBlock {
-			prestateRoot, err := topProvider.AbsolutePreStateCommitment(ctx)
-			if err != nil {
-				return nil, fmt.Errorf("failed to retrieve absolute prestate output root: %w", err)
-			}
-			agreed = contracts.Proposal{
-				L2BlockNumber: new(big.Int).SetUint64(topProvider.prestateBlock),
-				OutputRoot:    prestateRoot,
-			}
-		} else {
-			preBlockNum, err := topProvider.BlockNumber(pre.Position)
-			if err != nil {
-				return nil, fmt.Errorf("unable to calculate pre-claim block number: %w", err)
-			}
-			agreed = contracts.Proposal{
-				L2BlockNumber: new(big.Int).SetUint64(preBlockNum),
-				OutputRoot:    pre.Value,
-			}
+		localContext := CreateLocalContext(pre, post)
+		proposals, err := FetchProposals(ctx, topProvider, pre, post)
+		if err != nil {
+			return nil, err
 		}
-		postBlockNum, err := topProvider.BlockNumber(post.Position)
+		return creator(ctx, localContext, depth, proposals[0], proposals[1])
+	}
+}
+
+func FetchProposals(ctx context.Context, topProvider *OutputTraceProvider, pre types.Claim, post types.Claim) ([2]contracts.Proposal, error) {
+	usePrestateBlock := pre == (types.Claim{})
+	var agreed contracts.Proposal
+	if usePrestateBlock {
+		prestateRoot, err := topProvider.AbsolutePreStateCommitment(ctx)
 		if err != nil {
-			return nil, fmt.Errorf("unable to calculate post-claim block number: %w", err)
+			return [2]contracts.Proposal{}, fmt.Errorf("failed to retrieve absolute prestate output root: %w", err)
 		}
-		claimed := contracts.Proposal{
-			L2BlockNumber: new(big.Int).SetUint64(postBlockNum),
-			OutputRoot:    post.Value,
+		agreed = contracts.Proposal{
+			L2BlockNumber: new(big.Int).SetUint64(topProvider.prestateBlock),
+			OutputRoot:    prestateRoot,
 		}
-
-		return creator(ctx, localContext, depth, agreed, claimed)
+	} else {
+		preBlockNum, err := topProvider.BlockNumber(pre.Position)
+		if err != nil {
+			return [2]contracts.Proposal{}, fmt.Errorf("unable to calculate pre-claim block number: %w", err)
+		}
+		agreed = contracts.Proposal{
+			L2BlockNumber: new(big.Int).SetUint64(preBlockNum),
+			OutputRoot:    pre.Value,
+		}
+	}
+	postBlockNum, err := topProvider.BlockNumber(post.Position)
+	if err != nil {
+		return [2]contracts.Proposal{}, fmt.Errorf("unable to calculate post-claim block number: %w", err)
+	}
+	claimed := contracts.Proposal{
+		L2BlockNumber: new(big.Int).SetUint64(postBlockNum),
+		OutputRoot:    post.Value,
 	}
+	return [2]contracts.Proposal{agreed, claimed}, nil
```

Comment:
nit: Seems like it would be better to just make the return type `contracts.Proposal, contracts.Proposal, error` rather than use a fixed size array here.